packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -84,14 +91,42 @@ export default class BuildManifestPlugin {
           assetMap.pages['/'] = assetMap.pages['/index']
         }
 
+        // Add the runtime buildManifest.js file (generated later in this file) as a
+        // dependency for the app. If the flag is false, the file won't be downloaded
+        // by the client
+        if (this.options.clientManifest) {
+          assetMap.pages['/_app'].push(
+            `${CLIENT_STATIC_FILES_PATH}/${
+              this.options.buildId
+            }/_buildManifest.js`
+          )
+        }
+
         assetMap.pages = Object.keys(assetMap.pages)
           .sort()
           // eslint-disable-next-line
           .reduce((a, c) => ((a[c] = assetMap.pages[c]), a), {} as any)
 
+        const manifestContentString = JSON.stringify(assetMap, null, 2)
+
         compilation.assets[BUILD_MANIFEST] = new RawSource(
-          JSON.stringify(assetMap, null, 2)
+          manifestContentString
         )
+
+        const clientManifestPath = `${CLIENT_STATIC_FILES_PATH}/${
+          this.options.buildId
+        }/_buildManifest.js`
+        const clientManifestContentString = manifestContentString.replace(
```

Comment:
Is this a safe operation? Not sure if we'll ever support spaces in any of this content.

Can we `JSON.stringify` without `2` instead?